### PR TITLE
Allow version edits of build-tools

### PR DESCRIPTION
**Pull request by phadej**
```
parallels@ubuntu:~/mess/hackage-cli$ diff -u HTTP.cabal.orig HTTP.cabal.new
--- HTTP.cabal.orig	2016-01-25 09:15:08.364000000 +0200
+++ HTTP.cabal.new	2016-01-25 09:28:44.992000000 +0200
@@ -1,5 +1,6 @@
 Name: HTTP
 Version: 4000.3.1
+x-revision: 1
 Cabal-Version: >= 1.8
 Build-type: Simple
 License: BSD3
@@ -115,7 +116,7 @@
   else
     Build-depends: network >= 2.2.1.8 && < 2.6
 
-  build-tools: ghc >= 7.0 && < 7.12
+  build-tools: ghc >= 7.0 && < 8.1, alex
 
   if flag(warn-as-error)
     ghc-options:      -Werror
@@ -126,7 +127,7 @@
 Test-Suite test
   type: exitcode-stdio-1.0
 
-  build-tools: ghc >= 7.0 && < 7.12
+  build-tools: ghc >= 7.0 && < 8.1
 
   hs-source-dirs: test
   main-is: httpTests.hs
parallels@ubuntu:~/mess/hackage-cli$ stack exec hackage-cli -- check-revision HTTP.cabal.new HTTP.cabal.orig
change not allowed:
Cannot change which build tools are dependencies, just their version constraints.
```

```
parallels@ubuntu:~/mess/hackage-cli$ diff -u HTTP.cabal.orig HTTP.cabal.new2
--- HTTP.cabal.orig	2016-01-25 09:15:08.364000000 +0200
+++ HTTP.cabal.new2	2016-01-25 09:28:27.852000000 +0200
@@ -1,5 +1,6 @@
 Name: HTTP
 Version: 4000.3.1
+x-revision: 1
 Cabal-Version: >= 1.8
 Build-type: Simple
 License: BSD3
@@ -115,7 +116,7 @@
   else
     Build-depends: network >= 2.2.1.8 && < 2.6
 
-  build-tools: ghc >= 7.0 && < 7.12
+  build-tools: ghc >= 7.0 && < 8.1
 
   if flag(warn-as-error)
     ghc-options:      -Werror
@@ -126,7 +127,7 @@
 Test-Suite test
   type: exitcode-stdio-1.0
 
-  build-tools: ghc >= 7.0 && < 7.12
+  build-tools: ghc >= 7.0 && < 8.1
 
   hs-source-dirs: test
   main-is: httpTests.hs
parallels@ubuntu:~/mess/hackage-cli$ stack exec hackage-cli -- check-revision HTTP.cabal.new2 HTTP.cabal.orig
change allowed:
what: the library component's dependency on build tool ghc
 old: >=7.0 && <7.12
 new: >=7.0 && <8.1
what: the test suite 'test' component's dependency on build tool ghc
 old: >=7.0 && <7.12
 new: >=7.0 && <8.1
```

### CODE REVIEW

File: src/Distribution/Server/Util/CabalRevisions.hs [hunk 1]
```diff
@@ -272,14 +274,18 @@ checkDependencies componentName ds1 ds2 =
       . Map.fromListWith (flip intersectVersionRanges)
       . map (\(Dependency pkgname verrange) -> (pkgname, verrange))
 
-checkDependency :: ComponentName -> Check Dependency
-checkDependency componentName (Dependency pkgA verA) (Dependency pkgB verB)
+checkDependency :: ComponentName
+                -> Bool               -- ^ build-depends or build-tools
```
Comment (hvr) [inline src/Distribution/Server/Util/CabalRevisions.hs]: this seems like a case of "boolean blindless" :-)

Comment (phadej) [inline src/Distribution/Server/Util/CabalRevisions.hs]: I'll add a specific `data`

**Comment by hvr** [comment]
lgtm

**Comment by phadej** [comment]
Had to resolve the conflicts, removed different messages for build-tools and build-depends, as it's a bit tricker now. Is it still ok?

**Comment by phadej** [comment]
the `hackage-server` PR: https://github.com/haskell/hackage-server/pull/495

**Comment by phadej** [comment]
ping. The hackage-server patch is accepted (and even deployed?)

**Pull request merged**